artichoke-backend/src/intern.rs

```diff
@@ -76,56 +23,85 @@ impl Artichoke {
             Ok(None)
         }
     }
+
+    pub fn intern_bytes_with_trailing_nul<T>(&mut self, bytes: T) -> Result<u32, Exception>
+    where
+        T: Into<Cow<'static, [u8]>>,
+    {
+        let bytes = bytes.into();
+        let state = self.state.as_mut().ok_or(InterpreterExtractError)?;
+        let symbol = state.symbols.intern(bytes)?;
+        let symbol = u32::from(symbol);
+        // mruby expexts symbols to be non-zero.
+        let symbol = symbol
+            .checked_add(<Self as Intern>::SYMBOL_RANGE_START)
+            .ok_or_else(SymbolOverflowError::new)?;
+        Ok(symbol)
+    }
+
+    pub fn check_interned_bytes_with_trailing_nul(
+        &self,
+        bytes: &[u8],
+    ) -> Result<Option<u32>, Exception> {
+        let state = self.state.as_ref().ok_or(InterpreterExtractError)?;
+        let symbol = state.symbols.check_interned(bytes);
+        if let Some(symbol) = symbol {
+            let symbol = u32::from(symbol);
+            // mruby expexts symbols to be non-zero.
+            let symbol = symbol
+                .checked_add(<Self as Intern>::SYMBOL_RANGE_START)
+                .ok_or_else(SymbolOverflowError::new)?;
+            Ok(Some(symbol))
+        } else {
+            Ok(None)
+        }
+    }
 }
 
 impl Intern for Artichoke {
-    type Symbol = Symbol;
+    type Symbol = u32;
 
     type Error = Exception;
 
+    const SYMBOL_RANGE_START: Self::Symbol = 1;
```

Comment:
this change comes from the 0.3.0 bump in `artichoke-core` to support the `'static` `AllSymbols` iterator.

It also makes the symbol ID fiddling more explicit below by giving a name to the offset constant.